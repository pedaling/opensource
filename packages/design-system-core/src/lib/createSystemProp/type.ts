import type { Theme } from '../types';

export type InterpolationFn = (props: Record<string, any>) => Record<string, any>;

export type SystemPropConfig = {
  property: string;
  styleProperty?: string;
  scale?: keyof Pick<Theme, 'colors' | 'space' | 'opacity'>;
  disabled?: boolean;
  transform?: (value: any) => Record<string, any> | undefined;
};

export type SystemProp = {
  (props: Record<string, any>): Record<string, any>;
  propName: string;
  disabled: boolean;
};
