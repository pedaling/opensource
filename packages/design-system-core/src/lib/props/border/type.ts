import type { ResponsiveValue } from '../../types';

export type BorderProps = {
  borderWidth?: ResponsiveValue<number>;
  borderStyle?: ResponsiveValue<'solid' | 'none'>;
  borderColor?: ResponsiveValue<string>;
  borderTopWidth?: ResponsiveValue<number>;
  borderTopStyle?: ResponsiveValue<'solid' | 'none'>;
  borderTopColor?: ResponsiveValue<string>;
  borderRightWidth?: ResponsiveValue<number>;
  borderRightStyle?: ResponsiveValue<'solid' | 'none'>;
  borderRightColor?: ResponsiveValue<string>;
  borderBottomWidth?: ResponsiveValue<number>;
  borderBottomStyle?: ResponsiveValue<'solid' | 'none'>;
  borderBottomColor?: ResponsiveValue<string>;
  borderLeftWidth?: ResponsiveValue<number>;
  borderLeftStyle?: ResponsiveValue<'solid' | 'none'>;
  borderLeftColor?: ResponsiveValue<string>;
  borderRadius?: ResponsiveValue<number | string>;
  borderTopLeftRadius?: ResponsiveValue<number | string>;
  borderTopRightRadius?: ResponsiveValue<number | string>;
  borderBottomLeftRadius?: ResponsiveValue<number | string>;
  borderBottomRightRadius?: ResponsiveValue<number | string>;
  outlineWidth?: ResponsiveValue<number>;
  outlineStyle?: ResponsiveValue<'solid' | 'none'>;
  outlineColor?: ResponsiveValue<string>;
  outlineOffset?: ResponsiveValue<number>;
};
