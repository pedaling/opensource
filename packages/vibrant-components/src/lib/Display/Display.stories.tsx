import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { Display } from './Display';

export default {
  title: 'Display',
  component: Display,
  args: {
    level: 1,
    children: 'Sample Text',
  },
} as ComponentMeta<typeof Display>;

export const Basic: ComponentStory<typeof Display> = ({ children, ...restProps }) => (
  <Display {...restProps}>{children}</Display>
);
