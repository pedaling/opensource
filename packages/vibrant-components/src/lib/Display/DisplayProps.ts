import type { ResponsiveValue } from '@vibrant-ui/core';
import { propVariant, withVariation } from '@vibrant-ui/core';
import type { TextProps } from '../Text';

export type DisplayProps = Omit<TextProps, 'kind'> & {
  level: ResponsiveValue<1 | 2 | 3 | 4>;
};

export const withDisplayVariation = withVariation<DisplayProps>()(
  propVariant({
    props: [
      {
        name: 'level',
        responsive: true,
      },
    ],
    variants: ({ level }) => ({
      kind: `display${level}` as const,
    }),
  })
);
