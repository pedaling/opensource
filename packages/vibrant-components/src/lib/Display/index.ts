export { Display } from './Display';
export type { DisplayProps } from './DisplayProps';
