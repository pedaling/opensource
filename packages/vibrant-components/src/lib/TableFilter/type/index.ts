export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
