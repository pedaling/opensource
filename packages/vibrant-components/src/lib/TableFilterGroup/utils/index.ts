import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
