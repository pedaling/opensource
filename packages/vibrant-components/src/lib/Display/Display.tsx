import { Text } from '../Text';
import { withDisplayVariation } from './DisplayProps';

export const Display = withDisplayVariation(({ innerRef, children, ...textProps }) => (
  <Text ref={innerRef} {...textProps}>
    {children}
  </Text>
));
