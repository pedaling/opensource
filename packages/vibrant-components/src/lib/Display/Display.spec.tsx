import { baseTheme } from '@vibrant-ui/theme';
import type { ReactRenderer } from '@vibrant-ui/utils/testing-web';
import { createReactRenderer } from '@vibrant-ui/utils/testing-web';
import { Display } from './Display';

describe('<Display />', () => {
  const { render } = createReactRenderer();

  let renderer: ReactRenderer;

  let element: HTMLElement;

  describe('when Display with level 3 rendered', () => {
    beforeEach(async () => {
      renderer = render(
        <Display data-testid="display" level={3}>
          Hello World
        </Display>
      );

      element = await renderer.findByTestId('display');
    });

    it('display3 style applied', () => {
      const themeTypography = baseTheme.typography;

      expect(element).toHaveStyleRule('font-size', themeTypography.display3.fontSize);

      expect(element).toHaveStyleRule('line-height', themeTypography.display3.lineHeight);
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });
});
