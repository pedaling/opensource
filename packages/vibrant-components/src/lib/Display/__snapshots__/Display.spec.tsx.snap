// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> when Display with level 3 rendered match snapshot 1`] = `
.emotion-0 {
  font-family: Pretendard;
  font-size: 3rem;
  font-weight: 800;
  line-height: 3.75rem;
  color: #101010;
}

<div>
  <span
    class="emotion-0"
    data-testid="display"
  >
    Hello World
  </span>
</div>
`;
